pkgs/servers/dns/knot-dns/default.nix

```diff
@@ -63,6 +58,19 @@ stdenv.mkDerivation rec {
 
   passthru.tests = {
     inherit knot-resolver;
+
+    # Some dependencies are very version-sensitive, so the might get dropped
+    # or embedded after some update, even if the nixPackagers didn't intend to.
+    deps = runCommandLocal "knot-deps-test"
+      { nativeBuildInputs = [ (lib.getBin stdenv.cc.libc) ]; }
+      ''
+        for libname in libngtcp2 ${lib.optionalString stdenv.isLinux "libbpf"}; do
+          echo "Checking for $libname:"
+          ldd '${knot-dns.bin}/bin/knotd' | grep -F "$libname"
+          echo "OK"
+        done
+        touch "$out"
+      '';
```

Comment:
I am not really convinced on the usefulness of this.

Reply:
I don't feel strongly about that.  I wasn't sure, but I'd do similar checks manually on updates of `knot-dns` or these dependencies, so I thought why not automate them (ofBorg) and also make it easier for me (`nix build -f. knot-dns.tests`).  I don't expect they'd have... noticeable cons, even if the pros are minor.